### PR TITLE
Control Assist with local USB microphone

### PR DESCRIPTION
**Pull request by synesthesiam**
Allows controlling Assist with a local USB microphone attached to the Home Assistant server.

A [previous version](https://github.com/rhasspy/hassio-addons/tree/master/assist_microphone) of the add-on existed externally, but this version is significantly different:

- Based on the [Wyoming satellite](https://github.com/rhasspy/wyoming-satellite), so it can be discovered automatically and represented with an HA device
- Pipeline selection and sensors/switches are available for automation (e.g., disable satellite using a presence sensor)
- Delegates all audio processing to HA (noise suppression, etc.) instead of duplicating functionality
- Simplifies code base by not supporting UDP audio, different Linux audio systems, etc.

### CODE REVIEW

File: assist_microphone/config.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+options:
+  awake_wav: "/usr/src/sounds/awake.wav"
+  done_wav: "/usr/src/sounds/done.wav"
+  snd_volume_multiplier: 1.0
+  debug_logging: false
+schema:
+  awake_wav: str
+  done_wav: str
+  snd_volume_multiplier: float
+  debug_logging: bool
```
Comment (frenck) [inline assist_microphone/config.yaml:26]: Should we make a debug level option instead and tie it in with that?

File: assist_microphone/rootfs/etc/s6-overlay/s6-rc.d/assist_microphone/finish [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/command/with-contenv bashio
+# shellcheck shell=bash
+# ==============================================================================
+# Take down the S6 supervision tree when service fails
+# s6-overlay docs: https://github.com/just-containers/s6-overlay
+# ==============================================================================
+declare exit_code
```
Comment (frenck) [inline assist_microphone/rootfs/etc/s6-overlay/s6-rc.d/assist_microphone/finish:7]: Seems to be unused?

File: assist_microphone/rootfs/etc/s6-overlay/s6-rc.d/assist_microphone/run [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/command/with-contenv bashio
+# shellcheck shell=bash
+# ==============================================================================
+# Start service
+# ==============================================================================
+export PYTHONPATH=/usr/src
```
Comment (frenck) [inline assist_microphone/rootfs/etc/s6-overlay/s6-rc.d/assist_microphone/run:6]: Why is this needed?

File: assist_microphone/translations/en.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+---
+configuration:
+  awake_wav:
+    name: Awake WAV
+    description: >-
+      Path to WAV file to play when wake word is detected (empty to disable)
```
Comment (frenck) [inline assist_microphone/translations/en.yaml:6]:
```suggestion
      Path to WAV file to play when wake word is detected (empty to disable).
```

File: assist_microphone/translations/en.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  done_wav:
+    name: Done WAV
+    description: >-
+      Path to WAV file to play when voice command is finished (empty to disable)
```
Comment (frenck) [inline assist_microphone/translations/en.yaml:10]:
```suggestion
      Path to WAV file to play when voice command is finished (empty to disable).
```

File: assist_microphone/README.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+# Home Assistant Add-on: Assist Microphone
+
+![Supports aarch64 Architecture][aarch64-shield] ![Supports amd64 Architecture][amd64-shield] ![Supports armv7 Architecture][armv7-shield]
+
+Home Assistant add-on that uses a local USB microphone to control  [Assist](https://www.home-assistant.io/voice_control/).
```
Comment (frenck) [inline assist_microphone/README.md:5]:
```suggestion
Home Assistant add-on that uses a local USB microphone to control [Assist](https://www.home-assistant.io/voice_control/).
```

File: assist_microphone/build.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+---
+build_from:
+  amd64: ghcr.io/home-assistant/amd64-base-debian:bookworm
+  aarch64: ghcr.io/home-assistant/aarch64-base-debian:bookworm
+  armv7: ghcr.io/home-assistant/armv7-base-debian:bookworm
```
Comment (frenck) [inline assist_microphone/build.yaml:5]: Any specific reason we built this around Debian and not our default Alpine?

**Review by frenck — CHANGES_REQUESTED** [review]

File: assist_microphone/build.yaml (hunk 7 above)
Comment (synesthesiam) [reply assist_microphone/build.yaml]: I use the `webrtc-noise-gain` package for audio enhancements, which has a pre-compiled wheel for glibc.

File: assist_microphone/rootfs/etc/s6-overlay/s6-rc.d/assist_microphone/run (hunk 3 above)
Comment (synesthesiam) [reply assist_microphone/rootfs/etc/s6-overlay/s6-rc.d/assist_microphone/run]: Not needed anymore, thanks!

File: assist_microphone/rootfs/etc/s6-overlay/s6-rc.d/assist_microphone/finish (hunk 2 above)
Comment (synesthesiam) [reply assist_microphone/rootfs/etc/s6-overlay/s6-rc.d/assist_microphone/finish]: Removed. It's there in other `finish` scripts, so I've made a task to clean those up too.

File: assist_microphone/config.yaml (hunk 1 above)
Comment (synesthesiam) [reply assist_microphone/config.yaml:33]: Can you go into more detail about what you mean here?

Comment (frenck) [reply assist_microphone/config.yaml:33]: Let's keep it at this for now, and ship this. We can always improve this part if needed.

👍

**Review by frenck — APPROVED** [review]
Thanks, @synesthesiam 👍 

../Frenck
